web/tutorial-actions-executor/e2e-tests/generate-app.test.ts

```diff
@@ -0,0 +1,70 @@
+import path from "path";
+import { describe, expect, it } from "vitest";
+import { $, fs, glob } from "zx";
+
+const TEST_APP_NAME = "e2e-test-app";
+const TEST_OUTPUT_DIR = "./e2e-tests/.result";
+const TEST_TUTORIAL_DIR = "./e2e-tests/fixtures/tutorial";
+const WASP_CLI_COMMAND = process.env.WASP_CLI_COMMAND ?? "wasp-cli";
+
+describe("generate-app e2e", () => {
+  it(
+    "should generate app from tutorial actions and match snapshots",
+    async () => {
+      const generateAppOptions = [
+        "--app-name",
+        TEST_APP_NAME,
+        "--output-dir",
+        TEST_OUTPUT_DIR,
+        "--tutorial-dir",
+        TEST_TUTORIAL_DIR,
+        "--wasp-cli-command",
+        WASP_CLI_COMMAND,
+      ];
+      await $`npm run tacte -- generate-app ${generateAppOptions}`.verbose();
+
+      const projectDirPath = path.join(TEST_OUTPUT_DIR, TEST_APP_NAME);
+
+      function getProjectFileContent(filePath: string): string {
+        return fs.readFileSync(path.join(projectDirPath, filePath), "utf-8");
+      }
```

Comment:
minor, style: i'd bump this up to the top level scope

Reply:
+1, since another one is also outside, and I don't nee dto know this implementation here yet.